CMakeLists.txt

```diff
@@ -2,7 +2,10 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-cmake_minimum_required(VERSION 3.18)
+cmake_minimum_required(VERSION 3.20)
```

Comment:
The change is required so that we can enable CMP0118. That policy makes things much much easier when we want to set sources as GENERATED. Right now, that is not required, I mean... none of this is required, this function is not used anywhere :D

If we don't set the policy we would need to break a little bit the encapsulation of modules, because the GENERATED property would need to be set in multiple places. That said, let's cross that bridge when we get there. This was YOLO sprint work and it is highly possible there is some other way to handle GENERATED files thta allows us not to increase the CMake version.

Reply:
I'll take a stab at seeing what it takes to update CMake on the PPA. It might not be as hard as I think it is.

Reply:
The [ppa:okirby/qt6-backports](https://launchpad.net/~okirby/+archive/ubuntu/qt6-backports) should now have CMake 3.22 available, backported from Ubuntu 22.04/Jammy, so we can add this policy back to help with dependencies. Though, you will also need to rebase ontop of #9127 to make sure the taskcluster worker is updated.